src/utils.js

```diff
@@ -0,0 +1,189 @@
+const fsExtra = require('fs-extra')
+const markdownTable = require('markdown-table')
+const { rulesMap } = require('./rules-map')
+
+/**
+ * Expected-Vs-Obtained values of Auto-WCAG vs Axe-Core Results
+ * and the output to be reported
+ */
+const wcagVsAxeResultsMap = {
+  'passed-passes': 'automated',
+  'passed-violations': 'incorrect',
+  'passed-incomplete': 'semi-automated',
+  'passed-inapplicable': 'automated', // * issue - https://github.com/auto-wcag/auto-wcag/issues/265
+  'failed-passes': 'incorrect',
+  'failed-violations': 'automated',
+  'failed-incomplete': 'semi-automated',
+  'failed-inapplicable': 'incorrect',
+  'inapplicable-passes': 'automated', // * issue - https://github.com/auto-wcag/auto-wcag/issues/265
+  'inapplicable-violations': 'incorrect',
+  'inapplicable-incomplete': 'incorrect',
+  'inapplicable-inapplicable': 'automated'
+}
+
+/**
+ * Cell Style for the markdown table report type
+ */
+const ruleCellStatusStyle = {
+  automated: 'background-color: lightseagreen; color: black; padding: 4px;',
+  'semi-automated': 'background-color: orange; color: black; padding: 4px;',
+  incorrect: 'background-color: red; color: white; padding: 4px;'
+}
+
+/**
+ * Get style of cell based on rule status
+ * Convenience method to get a table cell style for markdown cell contents
+ * @param {String} status output result of a testcase, which is mapped to style of table cell
+ * @returns {String} Style based on rule status
+ */
+function getCellStyle(status) {
+  return ruleCellStatusStyle[status]
+}
+
+/**
+ * Get rule status
+ * @param {Array<Object>} ruleTestCasesResults results of all testcases per ruleId of Auto-WCAG
+ * @returns {String} rule status
+ */
+function getRuleStatus(ruleTestCasesResults) {
+  const anyIncorrect = ruleTestCasesResults.some(
+    result => result.testcaseStatus === 'incorrect'
+  )
+  if (anyIncorrect) {
+    return 'incorrect'
+  }
+  const anySemiAutomated = ruleTestCasesResults.some(
+    result => result.testcaseStatus === 'semi-automated'
+  )
+  if (anySemiAutomated) {
+    return 'semi-automated'
+  }
+  return 'automated'
+}
+
+/**
+ * Get testcase status
+ * @param {Object} param composite object with attributes to compute testcase status
+ * @property {Object} param.result result of testcase
+ * @property {Object} param.testcase testcase
+ * @returns {String} status of testcase
+ */
+function getTestCaseStatus({ result, testcase }) {
```

Comment:
This function should be part of the testrunner. The test engine should return its findings to the testrunner, the test runner should decide the status. The rules for deciding the status shouldn't change between implementors.